lib/ansible/module_utils/facts/hardware/linux.py

```diff
@@ -138,8 +138,16 @@ def get_memory_facts(self):
         if None not in (memstats.get('swaptotal'), memstats.get('swapfree')):
             memstats['swap:used'] = memstats['swaptotal'] - memstats['swapfree']
 
+        if os.path.isdir('/sys/devices/system/edac/mc'):
```

Comment:
this is not a reliable path, i would instead look at dmi facts

Reply:
I am unsure why you mean that is not reliable, kernel says in https://www.kernel.org/doc/Documentation/ABI/README 
```
	This directory documents interfaces that are felt to be stable,
	as the main development of this interface has been completed.
	The interface can be changed to add new features, but the
	current interface will not break by doing this, unless grave
	errors or security problems are found in them.  Userspace
	programs can start to rely on these interfaces, but they must be
	aware of changes that can occur before these interfaces move to
	be marked stable.
```
so yes, it may change again, but it's not very likely and would need to be for good reason.

So do you want me to build a fallback on dmidecode as it's here:
https://github.com/ansible/ansible/blob/a3ee846a646e8f3e15437f40e3c77bfcb02e0915/lib/ansible/module_utils/facts/hardware/linux.py#L372-L416 ?

Reply:
I think the comment about reliability means that it's often not populated with the data you are looking to get.  It actually doesn't actually contain the information on any system I have access to.  Relying on dmi over this path, would likely provide the information in more scenarios.

Reply:
^ exactly this, i've looked at 12 systems and only one had the data

Reply:
I think its only populated on hardware :) but yeah I can build a dmidecode fallback

Reply:
but with dmidecode this becomes a root-only operation, it's not with the sysfs files

Reply:
I changed it to include a dmidecode fallback, but now if it does not have root it will show 0 as physical mem instead of null, not sure which one is the better idea :)

Reply:
`N/A` and/or a warning would be more appropriate

Reply:
I can change that, lets see where the get_dmi_facts change request goes first and then I'll think about how to put this into the resulting code :)